src/nvim/eval.lua

```diff
@@ -13377,18 +13377,20 @@ M.funcs = {
     args = 1,
     base = 1,
     desc = [=[
-      The result is a Number, which is the width of window {nr}.
-      {nr} can be the window number or the |window-ID|.
-      When {nr} is zero, the width of the current window is
-      returned.  When window {nr} doesn't exist, -1 is returned.
-      An existing window always has a width of zero or more.
+      Gets the width of |window-ID| {nr} (zero for "current
+      window"), or -1 if window {nr} doesn't exist. An existing
+      window always has a width of zero or more.
+
       Examples: >vim
         echo "The current window has " .. winwidth(0) .. " columns."
         if winwidth(0) <= 50
           50 wincmd |
         endif
-      <For getting the terminal or screen size, see the 'columns'
-      option.
+      <
+      To get the visible "viewport", use |getwininfo()|: >vim
+          :echo getwininfo()[0].width - getwininfo()[0].textoff
```

Comment:
Why `[0]`?

Reply:
getwininfo() returns a List

Reply:
So why `[0]`?

Reply:
Yes, but why always use the first entry?

Reply:
```suggestion
          :echo getwininfo(winid)[0].width - getwininfo(winid)[0].textoff
```

Reply:
> Yes, but why always use the first entry?

Just trying to keep the example simple.